lib/core/Blockchain.ts

```diff
@@ -215,6 +215,13 @@ export default class Blockchain implements IBlockchain {
       return undefined;
     }
 
+    // Treat a HTTP 503 from the blockchain service as a non-error and
+    // return `undefined` such that a minimum-size batch will be
+    // written instead of the batch write being skipped
+    if (response.status === HttpStatus.SERVICE_UNAVAILABLE) {
+      return undefined;
+    }
+
     const responseBodyString = await ReadableStream.readAll(response.body);
     if (response.status !== HttpStatus.OK) {
       throw new SidetreeError(CoreErrorCode.BlockchainGetWriterLockResponseNotOk, `Status: ${response.status}. Response: ${responseBodyString}.`);
```

Comment:
Shouldn't this also result in minimum-size batch to written also?

Reply:
@thehenrytsai: Ah, sorry it did not register with me earlier that you meant to change the behaviour in response to _all_ non-200/404 status codes. My intent is to only change the behaviour in response to a 503, to be the same as to a 404.

We have a deployment in which a sort-of proxy will occasionally respond to requests to `${uri}/writerlock` with a 503. In the existing implementation of `Blockchain.ts`, this will result in an error being thrown. By the time the error makes it to the point in the call stack at which it is caught, it cannot be distinguished from other (actual) errors, and appears as a failure even though it isn't really.

So all I want to do is not treat a 503 as an error; other status codes are beyond the scope of this PR.

Reply:
I understand as I've seen that occasional behavior. = ]

At minimum you'll need to update the interface description with this new behavior since this is a deviation to existing contract.
```javascript
  /**
   * Gets the lock object required for batch writing.
   *
   * @returns the lock object if one exist; undefined otherwise.
   * @throws SidetreeError with ErrorCode.ValueTimeLockInPendingState if the lock is not yet confirmed on the blockchain.
   */
  getWriterValueTimeLock (): Promise<ValueTimeLockModel | undefined>;
```

But from interface design standpoint, it is rather odd that `undefined` is returned for one error but not for another, hence the suggestion to just written `undefined` for all errors as a compromise while keeping the interface clean.

Reply:
Hi @thehenrytsai, I don't mean to seem like I'm insisting on introducing any inconsistency. I'd happily abandon this PR if you could approve https://github.com/decentralized-identity/sidetree/pull/1203, instead; this would enable us to customise the behaviour of `Blockchain.ts` to accommodate our deployment environment without changing the semantics of the interface here.

Reply:
Hey @viathefalcon, you don't sounds like insistent at all, and seriously thanks for looking into any of this!

Your contribution and insights here are super valuable and thus I really want to figure out a happy middle ground to get this PR checked in so your work doesn't go to waste and everyone can benefit! But I also know your time is limited and valuable, so if you can't make further investment for this one I totally understand.

I have reviewed, approved, and merged your dependency injection PR to at least unblock you.